### ISSUE TITLE
wx.Image.Clear(value=0) fails with "argument 1 has unexpected type int"

### ISSUE THREAD

**Issue by pinhead216** [wxWidgets/Phoenix#2433]
**Operating system**: Windows 11 22H2
**wxPython version & source**:    4.2.1 via pip install
**Python version & source**:   Stock 3.11.4
**Description of the problem**:  

Clearing an empty wx.Image that was just created will fail with "argument 1 has unexpected type int", if you however omit the value= parameter (defaulting to '0') it won't fail
If wx.Image has been assigned an Image, passing a value to Clear() won't fail but the colour will always be black irrespactive of the value passed. wx.Image.Clear(value=255) will still result in an all black image.

<details>
  <summary>Code Example (click to expand)</summary>

```python
import wx

image = wx.Image(100, 100)
image.Clear(value=0)
```
Will fail

```python
import wx

image = wx.Image(100, 100)
image.Clear()
```
Won't fail

```python
import wx
......
img = wx.Image('c:\\Install\\cover.jpg', wx.BITMAP_TYPE_ANY)
img = img.Scale(180, 180)
img.Clear(128)
```
Won't fail but will not clear the image with the requested colour (just black)

</details>

**Comment by reticulatus** [comment]
I noticed that the wxWidgets documentation for wxImage::Clear shows the value parameter as _unsigned char_.

I wondered what the following code would do:

```
import wx
image = wx.Image(5, 5)
b = bytes([65])
image.Clear(value=b)
d = image.GetData()
print(len(d), d)
```

When I ran it, it didn't throw an exception, and output the following:
```
75 bytearray(b'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA')
```

5 x 5 x 3RGB values = 75.

[Tested using Python 3.10.6 + wxPython 4.2.1 gtk3 (phoenix) wxWidgets 3.2.2.1 on Linux Mint 21.1]

**Comment by swt2c** [comment]
Good find @reticulatus.  Indeed, the C++ API expects an unsigned char, so this method requires special handling with sip to make it accept a Python Int.  Should be fixed by #2434.
